src/app/view/components/_common/action-menu-button/action-menu-button.component.spec.ts

```diff
@@ -97,7 +109,40 @@ describe('ActionMenuButtonComponent', () => {
 
         actionButtonTooltip.nativeElement.click();
 
-        expect(spyHandleAction).toHaveBeenCalledWith(fakeActions[0], mockContentType, new MouseEvent(''));
+        expect(spyHandleAction).toHaveBeenCalledWith(mockContentType, new MouseEvent(''));
     });
 
+    it('should filter actions based on shouldShow field', () => {
+        const fakeActions: DotDataTableAction[] = [
+            {
+                menuItem: {
+                    icon: 'fa-trash',
+                    label: 'Remove',
+                    command: () => {}
+                },
+                shouldShow: () => false
+            },
+            {
+                menuItem: {
+                    command: () => {},
+                    icon: 'fa-edit',
+                    label: 'Edit'
+                },
+                shouldShow: () => false
+            },
+            {
+                menuItem: {
+                    command: () => {},
+                    icon: 'fa-add',
+                    label: 'Add'
+                },
+                shouldShow: () => true
+            }
+        ];
+
+        comp.actions = fakeActions;
+        comp.ngOnInit();
+
+        expect(comp.filteredActions.length).toEqual(1);
+    });
```

Comment:
I think we need a test where shouldShow return false just for some items and true for the others

Reply:
that is done in this test, I am asking if filteredActions is equals to 1, when the actions array is 3.